### PR TITLE
gui: Add listView

### PR DESCRIPTION
**Pull request by janezd**
Working with models is cleaner and usually more convenient than with lists of strings. This PR adds `gui.listView`, which is similar to `gui.listBox` but uses a model instead of list, so values can be instances of `Variable` instead of their names. The code is tested on the box plot widget.

This currently does not work with context settings. We need to modify them to store the `name` if the value as an instance of `Variable` and to restore the `Variable`. For the latter, we need a flag either in the definition of the setting or in the saved settings. @astaric, any recommendations?

### CODE REVIEW

File: Orange/widgets/gui.py [hunk 1]
```diff
@@ -1195,6 +1195,36 @@ def attributeItem(var):
     return attributeIconDict[var], var.name
 
 
+class ListViewWithSizeHint(QListView):
+    def __init__(self, *args, preferred_size=None, **kwargs):
+        super().__init__(*args, **kwargs)
```
Comment (kernc) [inline Orange/widgets/gui.py]: Given how one can pass a preferred size as a kwarg into QWidget constructor, e.g.
```py
QListView(parent, sizeHint=QSize(100, 80), **other_properties)
```
is this class necessary?

Yes, QSize needs to be imported.

Comment (janezd) [inline Orange/widgets/gui.py]: I suppose I didn't know about the `sizeHint` property. I'll remove the class. Thanks.

No, `QSize` doesn't need to be imported. The class is (was) intended to be used in `gui.listView`, which can accept a tuple and convert it.

Comment (kernc) [inline Orange/widgets/gui.py]: I meant in above case when setting sizeHint from kwarg. This makes it inconvenient compared to tuples, but is a rather small price to pay if it gets you rid of a whole class. :smiley:  

> "It is better to have 100 functions operate on one data structure than 10 functions on 10 data structures."
> — Alan Perlis

**Comment by ales-erjavec** [comment]
> Given how one can pass a preferred size as a kwarg into QWidget constructor, e.g.
> ```py
> QListView(parent, sizeHint=QSize(100, 80), **other_properties)
> ```
> is this class necessary?

That does not work as `sizeHint` is not a settable property. It might warrant a bug report to PyQt, though.

**Comment by kernc** [comment]
Ah, sorry. Thanks.

**Comment by astaric** [comment]
Regarding the ContextSettings, they should work. encode and decode convert between variables and serializable representation (name, type).

If they still do not work after rebasing this onto the latest master, I can take a look given the reproducible example.

**Pull request merged**